index.js

```diff
@@ -469,6 +476,28 @@ Router.prototype.use = function use(handler) {
     throw new TypeError('argument handler is required')
   }
 
+  var name
+
+  // If a name is used, the last argument will be a string, not a function
+  if (callbacks.length > 1 && typeof callbacks[callbacks.length - 1] !== 'function') {
```

Comment:
Can't we just shift them? I'm interested how this is going to be described in TypeScript :)